docs/rfc/rfc-9.adoc

```diff
@@ -0,0 +1,160 @@
+:toc: macro
+
+= RFC 9: Bitcoin Ordinals
+
+:icons: font
+:numbered:
+toc::[]
+
+== Background
+
+As of 2023-02-23, Bitcoin Ordinals are 🔥 HOT 🔥. Here is a
+https://insights.glassnode.com/ordinal-theory-and-the-rise-of-inscriptions/[decent
+explainer].
+
+image:assets/inscriptions-over-time.png[]
+
+There is not a good way to get those Ordinals over to Ethereum currently, where
+Ordinal owners can use the more mature NFT platforms. We could be that way!
+
+Here is an https://ordinals.com/inscription/cfa453dbf693641b8f0c6b83cb733e71832606fc26415c36c70744953d6f6161i0[example ordinal].
+
+=== Current Functionality
+
+Ordinals, at a high level, are single satoshis with metadata stuck into their
+witness field (called Inscribing). Each satoshi has an immutable age, which
+effectively gives them serial numbers.
+
+These Inscribed Satoshis (called Inscriptions) can be transferred like normal
+BTC. The sender's wallet software needs to be careful to not merge the
+Inscriptions with "normal" UTXOs, and needs to be careful to not accidentally
+send the Inscription as part of a payment. The
+https://github.com/casey/ord[casey/ord] repo provides the primatives for the
+handling of Ordinals on BTC.
+
+A good mental model here is thinking about Inscriptions like
+https://www.apmex.com/product/117942/1932-s-washington-quarter-choice-au[special
+edition quarters]. The linked quarter trades for ~$350, but is also legal tender
+worth $0.25. An owner of such a quarter probably does not want to use it to pay
+for parking. Inscriptions work the same way.
+
+Since Ordinals are still BTC, we can use most of the same tBTCv2 ideas for
+custodying Ordinals: rotating wallets, P2SH addresses, fraud proofs,
+redemptions, etc, with some slight modifications.
+
+== Proposal
+
+=== Overview
+
+We build a BTC <> Ethereum Ordinal bridge!
+
+We add functionality to the current tBTCv2 client to make future signing groups
+compatible with Ordinals. To deposit to these signing groups, we use the same
+P2SH process including a refund that we use for tBTCv2. They pay a <<Fees,fee>>.
+We optimistically mint using the same Minter & Guardian process we already
+havefootnote:[The structure for Minters and Guardians is the same, but the
+validation rules would be different]. In the background, we sweep to disable the
+refund. Later, someone brings back the NFT and a <<Fees,fee>>. We burn the NFT
```

Comment:
> For example, if we have 5 inputs with ordinals, we can't sweep them to one output.

yeah exactly

> Is it just about relocking to another output to disable refunds? It feels like for n inputs we always need to create n outputs when sweeping ordinals. For tBTC, n inputs are swept to 1 output.

Completely correct. We would still need to do some sort of sweeping transaction to disable the refund. That transaction would need to have as many outputs as inputs since we can't combine them.